lib/bitcoin/lock/LockResolver.ts

```diff
@@ -3,9 +3,19 @@ import BitcoinError from '../BitcoinError';
 import BitcoinOutputModel from '../models/BitcoinOutputModel';
 import BlockchainLockModel from '../../common/models/BlockchainLockModel';
 import ErrorCode from '../ErrorCode';
-import LockIdentifier from '../models/LockIdentifierModel';
+import LockIdentifierModel from '../models/LockIdentifierModel';
 import LockIdentifierSerializer from './LockIdentifierSerializer';
-import { Script, Address } from 'bitcore-lib';
+import { Script } from 'bitcore-lib';
+
+/** Structure (internal for this class) to hold the redeem script verification results */
+interface LockScriptVerifyResult {
+  /** whether or not the script was valid */
+  isScriptValid: boolean;
+  /** the target address when the script unlcoks; undefined if the script is not valid. */
+  publicKeyHashOut: string | undefined;

```

Comment:
Nit: I confused this as a script as I thought you mentioned that xxxOut typically refers to the "output script"? Such as `buildPublicKeyHashOut` and `targetScriptHashOut`.

```suggestion
  publicKeyHash: string | undefined;

```